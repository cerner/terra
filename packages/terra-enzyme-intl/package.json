{
  "name": "@cerner/terra-enzyme-intl",
  "version": "4.6.1",
  "description": "Enzyme helpers for fortifying tests that depend on react-intl by decoupling the need for actual translations.",
  "author": "Cerner Corporation",
  "main": "lib/index.js",
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/cerner/terra.git",
    "directory": "packages/terra-enzyme-intl"
  },
  "bugs": {
    "url": "https://github.com/cerner/terra/issues"
  },
  "homepage": "https://github.com/cerner/terra/packages/terra-enzyme-intl#readme",
  "publishConfig": {
    "access": "public"
  },
  "license": "Apache-2.0",
  "keywords": [
    "Cerner",
    "enzyme",
    "i18n",
    "intl",
    "jest",
    "react",
    "react-intl",
    "Terra",
    "terra-enzyme-intl"
  ],
  "files": [
    "lib"
  ],
  "dependencies": {
    "prop-types": "^15.5.8"
  },
  "devDependencies": {
    "react-intl": "^5.25.1"
  },
  "peerDependencies": {
    "enzyme": "3.x",
    "react": "16.x",
    "react-intl": "^5.25.1"
  },
  "scripts": {
    "clean:install": "npm run clean && npm install",
    "compile": "babel --root-mode upward src --out-dir lib --copy-files",
    "jest": "jest --config ../../jest.config.js",
    "lint": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
    "precompile": "rm -rf lib",
    "test": "npm run lint && npm run jest"
  }
}
