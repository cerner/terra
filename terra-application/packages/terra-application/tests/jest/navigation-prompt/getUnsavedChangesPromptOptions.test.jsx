import { mockIntl } from '@cerner/terra-enzyme-intl';

import getUnsavedChangesPromptOptions from '../../../src/navigation-prompt/getUnsavedChangesPromptOptions';

describe('getUnsavedChangesPromptOptions', () => {
  it('generates a function that provides appropriate messaging for a single prompt', () => {
    const generatedFunction = getUnsavedChangesPromptOptions(mockIntl);

    expect(generatedFunction).toBeDefined();

    const functionOutput = generatedFunction([{ description: 'Test Prompt' }]);

    expect(functionOutput).toMatchSnapshot();
  });

  it('generates a function that provides appropriate messaging for multiple prompts', () => {
    const generatedFunction = getUnsavedChangesPromptOptions(mockIntl);

    expect(generatedFunction).toBeDefined();

    const functionOutput = generatedFunction([{ description: 'Test Prompt' }, { description: 'Another Test Prompt' }]);

    expect(functionOutput).toMatchSnapshot();
  });
});
