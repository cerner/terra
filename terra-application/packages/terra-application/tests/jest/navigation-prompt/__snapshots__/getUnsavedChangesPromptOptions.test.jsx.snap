// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUnsavedChangesPromptOptions generates a function that provides appropriate messaging for a single prompt 1`] = `
Object {
  "acceptButtonText": "terraApplication.unsavedChangesPrompt.acceptButtonText",
  "buttonOrder": "rejectFirst",
  "content": undefined,
  "emphasizedAction": "reject",
  "endMessage": "terraApplication.unsavedChangesPrompt.multiplePromptMessageOutro",
  "rejectButtonText": "terraApplication.unsavedChangesPrompt.rejectButtonText",
  "startMessage": "terraApplication.unsavedChangesPrompt.singlePromptMessage",
  "title": "terraApplication.unsavedChangesPrompt.title",
}
`;

exports[`getUnsavedChangesPromptOptions generates a function that provides appropriate messaging for multiple prompts 1`] = `
Object {
  "acceptButtonText": "terraApplication.unsavedChangesPrompt.acceptButtonText",
  "buttonOrder": "rejectFirst",
  "content": <ContentLayoutAsList
    items={
      Array [
        "Test Prompt",
        "Another Test Prompt",
      ]
    }
  />,
  "emphasizedAction": "reject",
  "endMessage": "terraApplication.unsavedChangesPrompt.multiplePromptMessageOutro",
  "rejectButtonText": "terraApplication.unsavedChangesPrompt.rejectButtonText",
  "startMessage": "terraApplication.unsavedChangesPrompt.multiplePromptMessageIntro",
  "title": "terraApplication.unsavedChangesPrompt.title",
}
`;
